matlab/FeatherReader.cpp

```diff
@@ -0,0 +1,301 @@
+      break;
+    }
+    case ::arrow::StatusCode::IOError: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:IOError", arrow_error_message,
+                        status.ToString().c_str());
+      break;
+    }
+    case ::arrow::StatusCode::CapacityError: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:CapacityError",
+                        arrow_error_message, status.ToString().c_str());
+      break;
+    }
+    case ::arrow::StatusCode::UnknownError: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:UnknownError",
+                        arrow_error_message, status.ToString().c_str());
+      break;
+    }
+    case ::arrow::StatusCode::NotImplemented: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:NotImplemented",
+                        arrow_error_message, status.ToString().c_str());
+      break;
+    }
+    case ::arrow::StatusCode::SerializationError: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:SerializationError",
+                        arrow_error_message, status.ToString().c_str());
+      break;
+    }
+    case ::arrow::StatusCode::PythonError: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:PythonError",
+                        arrow_error_message, status.ToString().c_str());
+      break;
+    }
+    case ::arrow::StatusCode::PlasmaObjectExists: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:PlasmaObjectExists",
+                        arrow_error_message, status.ToString().c_str());
+      break;
+    }
+    case ::arrow::StatusCode::PlasmaObjectNonexistent: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:PlasmaObjectNonexistent",
+                        arrow_error_message, status.ToString().c_str());
+      break;
+    }
+    case ::arrow::StatusCode::PlasmaStoreFull: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:PlasmaStoreFull",
+                        arrow_error_message, status.ToString().c_str());
+      break;
+    }
+    case ::arrow::StatusCode::PlasmaObjectAlreadySealed: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:PlasmaObjectAlreadySealed",
+                        arrow_error_message, status.ToString().c_str());
+      break;
+    }
+    default: {
+      mexErrMsgIdAndTxt("MATLAB:feather:read:arrow:status:UnknownStatus",
+                        arrow_error_message, "Unknown status");
+      break;
+    }
+  }
+}
+
+// Read the name of variable i from the Feather file as a mxArray*.
+mxArray* FeatherReader::ReadVariableName(
+    const std::shared_ptr<::arrow::Column>& column) const {
+  return mxCreateString(column->name().c_str());
+}
+
+template <::arrow::Type::type ArrowTypeID>
+mxArray* FeatherReader::ReadNumericVariableData(
+    const std::shared_ptr<::arrow::Column>& column) const {
+  typedef typename FeatherType<ArrowTypeID>::ArrowArrayType ArrowArrayType;
+  typedef typename FeatherType<ArrowTypeID>::MatlabType MatlabType;
```

Comment:
Rather than contaminate this function with Feather-specific details, why not use the Arrow types? `arrow::ipc::FeatherReader` returns plain old Arrow data structures